src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -653,10 +663,25 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
            method_replacement.wrapper_method.type_name, ".",
            method_replacement.wrapper_method.method_name, "() ",
            wrapper_method_ref);
+
+      modified = true;
+      current_instruction_swapped = true;
+      // No point in checking the other method wrappers
+      break;
+    }
+
+    if (current_instruction_swapped) {
+      last_instr = pInstr;
     }
   }
 
   if (modified) {
+    if (last_instr != nullptr) {
+      auto pSkipTailCallInstr = rewriter.NewILInstr();
+      pSkipTailCallInstr->m_opcode = CEE_NOP;
+      rewriter.InsertAfter(last_instr, pSkipTailCallInstr);
```

Comment:
This is where we should try to insert the void no-op method.
We have the last_instr we instrumented and we know it's the last thing before the `CEE_RET` instruction.